net/http/authn/context.go

```diff
@@ -1,14 +1,27 @@
 package authn
 
-import "context"
+import (
+	"context"
+	"crypto/x509"
+)
 
 type key int
 
 const (
 	tokenKey key = iota
 	localhostKey
+	x509CertsKey
 )
 
+// X509Certs returns the cert stored in the context, if it exists.
+func X509Certs(ctx context.Context) []*x509.Certificate {
+	c, ok := ctx.Value(x509CertsKey).([]*x509.Certificate)
+	if !ok {
+		return []*x509.Certificate{}
```

Comment:
Nit: since c gets the zero value of a failed type assertion using the comma-ok form, we can just return c (which will be nil).

```go
c, _ := ctx.Value(x509CertsKey).([]*x509.Certificate)
return c
```